### PR TITLE
DR-1788 - Add retry around non-transactional lookup of firestore entries

### PR DESCRIPTION
**Pull request by snf2ye**
- Add retry
- Add logging around the retry

I have tested snapshot create on my dev deployment that is currently setup to act like prod.

### CODE REVIEW

**Review by nmalfroy — APPROVED** [review]
Looks great!  At some point, we should move the retry logic into it's own utility method since I think we have a few different ways that we're implementing it but this is good

File: src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java [hunk 1]
```diff
@@ -623,16 +625,22 @@ void updateDirectoryEntry(Firestore firestore, String collectionId, FireStoreDir
...
-            ApiFuture<DocumentSnapshot> docSnapFuture = docRef.get();
-            return docSnapFuture.get();
-        } catch (AbortedException | ExecutionException ex) {
-            throw fireStoreUtils.handleExecutionException(ex, "lookupByPathNoXn");
+        RuntimeException lastException = null;
+        for (int retryNum = 0; retryNum < LOOKUP_RETRIES; retryNum++) {
+            try {
+                DocumentReference docRef =
+                    firestore
+                        .collection(collectionId)
```
Comment (nmalfroy) [inline src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java]: Academic question: I wonder if calling collection actually makes a request to firestore or if it just preps the request...if it's the prior, we may want to move that out of the retry loop

**Review by samanehsan — APPROVED** [review]

File: src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java (hunk 1 above)
Comment (snf2ye) [reply src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java]: Good point! I moved it out of the loop and re-ran my manual test which showed this work on retry.

Comment (nmalfroy) [reply src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java]: Ok cool!

**Pull request merged**